packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -137,7 +137,7 @@ class SelectedContent {
 ///  * [SelectionArea], which provides an overview of selection system.
 mixin Selectable implements SelectionHandler {
   /// {@macro flutter.rendering.RenderObject.getTransformTo}
-  Matrix4 getTransformTo(RenderObject? ancestor);
+  Matrix4 getTransformTo(RenderObject? target, { bool skipIfInUnpaintedSubtree = false });
```

Comment:
For my own understanding: Why do we need to modify the signature on Selectable as well? Looks like nothing in this PR is actually calling this with the new parameter? 

(also, this seems to be uncovered by tests if we need to keep it)

Reply:
`Selectable.getTransformTo` isn't a valid override of `RenderObject.getTransformTo` if a `RenderObject` uses that mixin. Alternatively I could change `Selectable` to an abstract class?

Reply:
That's kinda unfortunate that Selectable and RenderObject interact that way when used together. It's not immediately clear to me why the Selectable interface (when looked at it stand-alone) would need this parameter. Should we have named this method something else (and make implementors forward it to the RenderObject's getTransformTo, if that's what they want to use)?

Or does the caller of this method on Selectable actually need to care about unpainted subtrees and should use this parameter?

@chunhtai Do you have any thoughts on this?

> Alternatively I could change Selectable to an abstract class?

How would you then make a RenderObject subclass use this?

Reply:
There's no implementation in the `Selectable` mixin so I can turn that into an abstract class? It could be breaking since people would have to replace `with` with `implements`

Reply:
Yes, but wouldn't `class Foo extends RenderObject implements Selectable` still have the problem that you cannot satisfy both overrides?

Reply:
If this is abstract class and this method have a different signature, they would have to manually wire it up when they create a renderobject that implements this abstract class.

what is the use case for `skipIfInUnpaintedSubtree: false`?

Reply:
> wouldn't class Foo extends RenderObject implements Selectable still have the problem that you cannot satisfy both overrides?

According to dartpad that works. My guess is `with Selectable` creates a synthetic class that tries to override `getTransformTo(RenderObject, { bool ....})` with `getTransformTo(RenderObject)`.

> what is the use case for skipIfInUnpaintedSubtree: false

idk, maybe in an animation callback where a `RenderOpacity`'s `paintsChild` returns false but will return true once the render object updates its layout in that frame?

Reply:
I would probably remove skipIfInUnpaintedSubtree and don't worry about the signature mismatch. It doesn't seem worth it to modify RenderObject API without a solid use case.

I am ok if we convert to abstract class, this seems like an oversight when i implement this class. but it doesn't seem to help much?